packages/core/src/_common/base/cancellable_promise.js

```diff
@@ -0,0 +1,27 @@
+export const makeCancelablePromise = listener => {
```

Comment:
Can the function name be consistent with the file name?

```suggestion
export const makeCancellablePromise = listener => {
```